File: java/bridge/eje1/EncryptedMessage.java
package patrones.bridge.eje1;

/**
 * 
 * EncryptedMessage encripta un mensaje entrante al trasladar todos los
 * caracteres a la derecha por una posición.
 * 
 */
public class EncryptedMessage implements Message {
	private MessageLogger logger;

	public EncryptedMessage(MessageLogger l) {
		logger = l;
	}

	public void log(String msg) {
		String str = preProcess(msg);
		logger.logMsg(str);
	}

	private String preProcess(String msg) {
		msg = msg.substring(msg.length() - 1)
				+ msg.substring(0, msg.length() - 1);
		return msg;
	}
}

File: c#/bridge/Ejm1/EncryptedMessage.cs
using System;
namespace Bridge.Ejm1
{
    /// <summary>
    /// EncryptedMessage encripta un mensaje entrante al trasladar todos los
    /// caracteres a la derecha por una posicion.
    /// </summary>
    public class EncryptedMessage : Message
    {
        private MessageLogger Logger;

        public EncryptedMessage(MessageLogger l)
        {
            Logger = l;
        }

        public void Log(String msg)
        {
            String str = PreProcess(msg);
            Logger.LogMsg(str);
        }

        private String PreProcess(String msg)
        {
            msg = msg.Substring(msg.Length - 1)
                    + msg.Substring(0, msg.Length - 1);
            return msg;
        }
    }
}
